### PR TITLE
Swift package support

### PR DESCRIPTION
**Pull request by grangej**
Wanted to contribute this back , this adds Swift Package support to the library.

### CODE REVIEW

**Comment by grangej** [comment]
I added a simple test case, I am unable to use the existing test cases since they contain mixed source, this is not supported under swift package manager.

File: Source/GCD/Include/CocoaAsyncSocket.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+@import Foundation;
+
+//! Project version number for CocoaAsyncSocket.
+FOUNDATION_EXPORT double cocoaAsyncSocketVersionNumber;
+
+//! Project version string for CocoaAsyncSocket.
+FOUNDATION_EXPORT const unsigned char cocoaAsyncSocketVersionString[];
+
+#import "../GCDAsyncSocket.h"
```
Comment (chrisballinger) [inline Source/GCD/Include/CocoaAsyncSocket.h]: This might cause problems. Maybe we should reorganize the folder structure instead?

**Comment by dsmurfin** [comment]
Out of interest why is there a restriction to macOS 10.14 in your manifest @grangej ? I'm working on a package myself and I'm currently linking to your fork.

**Comment by yvbeek** [comment]
You can use this fork of CocoaAsyncSocket for SwiftPM support if you like:
https://github.com/Building42/CocoaAsyncSocket

**Comment by chrisballinger** [comment]
SwiftPM support added in #706

**Pull request closed without merging**